src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -284,12 +284,20 @@ type RemoveDeletingObjectKeys<T> = RemoveNever<
   }
 >;
 
-type MergeObjectProperty<T, U, M extends MergeObjectOptions> = T extends Record<string, any>
+type MergeObjectProperty<T, U, M extends MergeObjectOptions> = M extends { enforceTypes: true }
+  ? T extends Array<any>
+    ? U extends Array<any>
+      ? U
+      : T
```

Comment:
I think this should be `never`?
```
mergeObject({a:["foo"]}, {a:"not an array"}, {enforceTypes: true})
Uncaught Error: Mismatched data types encountered during object merge.
```

Reply:
I *think* you might want something like this?
```typescript
type MergeObjectPropertyEnforce<
    T,
    U,
    M extends MergeObjectOptions,
> = T extends Array<any>
    ? U extends Array<any>
        ? U
        : never
    : T extends Record<string, any>
    ? U extends Record<string, any>
        ? Result<
              T,
              U,
              Omit<M, "insertKeys"> & { insertKeys: M["insertValues"] }
          >
        : never
    : U extends T
    ? U
    : never;

type MergeObjectPropertyLax<
    T,
    U,
    M extends MergeObjectOptions,
> = T extends Array<any>
    ? U
    : T extends Record<string, any>
    ? U extends Record<string, any>
        ? Result<
              T,
              U,
              Omit<M, "insertKeys"> & { insertKeys: M["insertValues"] }
          >
        : U
    : U;

type MergeObjectProperty<T, U, M extends MergeObjectOptions> = M extends {
    enforceTypes: true;
}
    ? MergeObjectPropertyEnforce<T, U, M>
    : MergeObjectPropertyLax<T, U, M>;
```

Reply:
I guess it won't be so easy, cause it will make `foundry.utils.mergeObject({ a: ['foo'] }, { a: { b: 'foo' } }, { enforceTypes: true })` to `{a: never}` instead of `never` :-/

Reply:
Perhaps this can be easier achieved with overloading. Will try later.

Reply:
Well the never deep inside should hopefully trigger a type error like the array did? I don't know how you could propagate it to the top